### PR TITLE
Enhancements

### PR DESCRIPTION
**Pull request by Lukkra211**

### CODE REVIEW

File: src/lightblue/entity.py [hunk 1]
```diff
@@ -202,3 +202,36 @@ def find_all(self, projection=None):
         lightblue_data['projection'] = projection
         return self.service.find_data(self.entity_name, self.version,
                                       lightblue_data)
+
+    def find_paginated(self, page_size, find_func,
+                       find_args=None, find_kwargs=None):
```
Comment (Allda) [inline src/lightblue/entity.py]: Why don't you use just *args and **kwargs?

Comment (Lukkra211) [reply src/lightblue/entity.py]: Fixed. I also refactored the method, since it was a bit hairy.

File: src/lightblue/service.py [hunk 2]
```diff
@@ -24,8 +24,8 @@ def __init__(
         ssl_verify=True,
         custom_session=None,
     ):
-        self.data_url = data_url
-        self.metadata_url = metadata_url
+        self.data_url = data_url.rstrip('/')
```
Comment (Allda) [inline src/lightblue/service.py:27]: No need to do that if you use urljoin

Comment (Lukkra211) [reply src/lightblue/service.py:27]: I tried but I am not sure if using urljoin is a good idea. Let's say I use it to format the url for a request:
```python
data_url1 = 'http://fake.lb.com/rest/data'
data_url2 = 'http://fake.lb.com/rest/data/'

path = 'insert/entity_name/version'

url1 = urlparse.urljoin(data_url1, path)  # http://fake.lb.com/rest/insert/entity_name/version
url2 = urlparse.urljoin(data_url2, path)  # http://fake.lb.com/rest/data/insert/entity_name/version

assert url1 != url2
```
Then it breaks the code for users who don't have the trailing slash at the end.

Comment (Lukkra211) [reply src/lightblue/service.py:27]: @Allda ^

File: src/lightblue/selection.py [hunk 3]
```diff
@@ -223,3 +231,40 @@ def all(self):
...
+            removed_fields (list):
+                list of sorted fields, for example:
+                ['field.3', 'field.1']
+
+        Returns:
+            dict: response from lightblue update query
+            int: count on removed fields, 0 if update failed
+
+        """
+        if len(fields) == 0:
```
[the remark this answers is not in the document]
Comment (Lukkra211) [reply src/lightblue/selection.py]: Ahh, didn't notice.

In the end, I decided that `if not fields` would be better for it's pythonic and doesn't require additional variable to be created.

File: src/lightblue/service.py (hunk 2 above)
Comment (Allda) [reply src/lightblue/service.py:27]: Aha ok, then it probably makes sense

**Pull request merged**